### PR TITLE
Refactor OpenID check behaviour

### PR DESCRIPTION
**Pull request by jaywink**
Default behaviour has been changed to verify any OpenID tokens of any homeserver. This keeps the code, configuration and examples simpler by reversing to requiring configuring a single homeserver for OpenID verifications explicitly.

Additionally this allows for the server and client Matrix API's to be on different domains, which was not true before when running against a single homeserver.

This is a breaking change which has been highlighted in the changelog.

This PR also fixes verifying tokens against homeservers which use SRV to delegate to another domain.

### CODE REVIEW

**Comment by jaywink** [comment]
This is btw already deployed to FOSDEM as a `v2.0.0-dev1` image and confirmed to work as intended.

File: src/verify.js [hunk 1]
```diff
@@ -78,47 +78,60 @@ function sanityCheckRequest(req, res, fields=[]) {
...
         }
+        serverName = process.env.UVS_OPENID_VERIFY_SERVER_NAME;
     } else {
-        homeserverUrl = process.env.UVS_HOMESERVER_URL;
+        serverName = req.body.matrix_server_name;
+    }
+    try {
+        homeserver = await matrixUtils.discoverHomeserverUrl(serverName);
+    } catch (error) {
+        logger.log('debug', `Failed to discover homeserver URL: ${error}`, {requestId: req.requestId});
```
Comment (Half-Shot) [inline src/verify.js:100]: should these be debug?

Comment (Half-Shot) [inline src/verify.js]: should these be a warn?

File: src/verify.js [hunk 2]
```diff
@@ -78,47 +78,60 @@ function sanityCheckRequest(req, res, fields=[]) {
...
-                logger.log(
-                    'warn',
-                    `Matrix user ID ${response.data.sub} from OpenID userinfo lookup does not ` +
-                        `match given matrix_server_name ${req.body.matrix_server_name}`,
-                    {requestId: req.requestId},
-                );
-                return false;
-            }
+        // Ensure the user ID actually matches the server name we checked against
+        if (!response.data.sub.endsWith(`:${serverName}`)) {
```
Comment (Half-Shot) [inline src/verify.js]: Check that this is a string?

**Review by Half-Shot — APPROVED** [review]
LGTM

**Pull request merged**